gst/nnstreamer/tensor_if/gsttensorif.c

```diff
@@ -1050,19 +1052,23 @@ gst_tensor_if_check_condition (GstTensorIf * tensor_if, GstBuffer * buf,
       }
       in_tensors[i].data = in_info[i].data;
       in_tensors[i].size = in_info[i].size;
-      gst_memory_unmap (in_mem[i], &in_info[i]);
     }
 
-    return tensor_if->custom.func (&tensor_if->in_config.info, in_tensors,
+    ret = tensor_if->custom.func (&tensor_if->in_config.info, in_tensors,
         tensor_if->custom.data, result);
+
+    for (i = 0; i < tensor_if->in_config.info.num_tensors; i++)
+      gst_memory_unmap (in_mem[i], &in_info[i]);
```

Comment:
```suggestion
for (i = 0; i < tensor_if->in_config.info.num_tensors; i++) {
      in_mem[i] = gst_buffer_peek_memory (buf, i);
      if (gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ)) {
        in_tensors[i].data = in_info[i].data;
        in_tensors[i].size = in_info[i].size;
        gst_memory_unmap (in_mem[i], &in_info[i]);     
      } else {
        GST_WARNING_OBJECT (tensor_if, "Cannot map input memory buffer(%u)\n",
            i);
        return FALSE;
      }

    ret = tensor_if->custom.func (&tensor_if->in_config.info, in_tensors,
        tensor_if->custom.data, result);
```

Reply:
tensor_if->custom.func() uses map-info (in_tensors[i].data = in_info[i].data) and this may not be valid after calling gst_memory_unmap().

Reply:
@jaeyun-jung Ah....right; Never mind.